src/module.py

```diff
@@ -3,22 +3,157 @@
 for preprocessing speech and text, the transformer & RNN encoder/decoder, & post
 processing modules for text and speech.
 '''
-# TODO: Consider adding pre/post net base class.
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import numpy as np
+from collections import OrderedDict
+
+from data.symbols import symbols
+
+class Linear(nn.Module):
+    """
+    Linear Module copied from Transformer-TTS.
+    
+    A linear layer initialized using the Xavier Uniform method.
+    """
+    def __init__(self, in_dim, out_dim, bias=True, w_init='linear'):
+        """
+        :param in_dim: dimension of input
+        :param out_dim: dimension of output
+        :param bias: boolean. if True, bias is included.
+        :param w_init: str. weight inits with xavier initialization.
+        """
+        super(Linear, self).__init__()
+        self.linear_layer = nn.Linear(in_dim, out_dim, bias=bias)
+
+        nn.init.xavier_uniform_(
+            self.linear_layer.weight,
+            gain=nn.init.calculate_gain(w_init))
+
+    def forward(self, x):
+        return self.linear_layer(x)
+
+class Conv(nn.Module):
+    """
+    Convolution Module copied from Transformer-TTS.
+    
+    A 1D convolution layer initialized using the Xavier Uniform
+    method.
+    """
+    def __init__(self, in_channels, out_channels, kernel_size=1, stride=1,
+                 padding=0, dilation=1, bias=True, w_init='linear'):
+        """
+        :param in_channels: dimension of input
+        :param out_channels: dimension of output
+        :param kernel_size: size of kernel
+        :param stride: size of stride
+        :param padding: size of padding
+        :param dilation: dilation rate
+        :param bias: boolean. if True, bias is included.
+        :param w_init: str. weight inits with xavier initialization.
+        """
+        super(Conv, self).__init__()
+
+        self.conv = nn.Conv1d(in_channels, out_channels,
+                              kernel_size=kernel_size, stride=stride,
+                              padding=padding, dilation=dilation,
+                              bias=bias)
+
+        nn.init.xavier_uniform_(
+            self.conv.weight, gain=nn.init.calculate_gain(w_init))
+
+    def forward(self, x):
+        x = self.conv(x)
+        return x
 
 class SpeechPrenet(nn.Module):
-    # TODO: Fill in from TTS repo :) 
-    def __init__(self):
+    """
+    Prenet for Speech Encoder copied from Transformer-TTS.
+    
+    As described in Ren's paper, 2-layer dense-connected network
+    with hidden size of 256, and the output dimension equals to
+    the hidden size of Transformer. Note that the dropout is
```

Comment:
can we add dropout?

Reply:
Yeah, this will be added back in. It's in the my other branch, I'll make a PR now,